### PR TITLE
fix(resource): fully handle default ack options

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This PR intends to fully handle default ACK options that can be set on Administration -> Parameters -> Monitoring

image

(ADDED: monitoring_default_acknowledgement_notify, monitoring_default_acknowledgement_with_services, monitoring_default_acknowledgement_force_active_checks)

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Reach JIRA ticket for details

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Application/Controller/Administration/ParametersController.php [hunk 1]
```diff
@@ -114,6 +126,15 @@ public function getParameters(): View
                 case self::DEFAULT_ACKNOWLEDGEMENT_STICKY:
                     $isAcknowledgementSticky = (int) $option->getValue() === 1 ? true : false;
                     break;
+                case self::DEFAULT_ACKNOWLEDGEMENT_NOTIFY:
+                    $isAcknowledgementNotify = (int) $option->getValue() === 1 ? true : false;
```
Comment (kduret) [inline src/Centreon/Application/Controller/Administration/ParametersController.php:130]:
```suggestion
                    $isAcknowledgementNotify = (int) $option->getValue() === 1;
```

File: src/Centreon/Application/Controller/Administration/ParametersController.php [hunk 2]
```diff
@@ -114,6 +126,15 @@ public function getParameters(): View
...
+                    break;
+                case self::DEFAULT_ACKNOWLEDGEMENT_WITH_SERVICES:
+                    $isAcknowledgementWithServices = (int) $option->getValue() === 1 ? true : false;
```
Comment (kduret) [inline src/Centreon/Application/Controller/Administration/ParametersController.php:133]:
```suggestion
                    $isAcknowledgementWithServices = (int) $option->getValue() === 1;
```

File: src/Centreon/Application/Controller/Administration/ParametersController.php [hunk 3]
```diff
@@ -114,6 +126,15 @@ public function getParameters(): View
...
+                    break;
+                case self::DEFAULT_ACKNOWLEDGEMENT_FORCE_ACTIVE_CHECKS:
+                    $isAcknowledgementForceActiveChecks = (int) $option->getValue() === 1 ? true : false;
```
Comment (kduret) [inline src/Centreon/Application/Controller/Administration/ParametersController.php:136]:
```suggestion
                    $isAcknowledgementForceActiveChecks = (int) $option->getValue() === 1;
```

File: src/Centreon/Application/Controller/Administration/ParametersController.php [hunk 4]
```diff
@@ -84,15 +90,21 @@ public function getParameters(): View
         $refreshInterval = '';
         $isAcknowledgementPersistent = false;
         $isAcknowledgementSticky = false;
+        $isAcknowledgementNotify = false;
+        $isAcknowledgementWithServices = false;
```
Comment (kduret) [inline src/Centreon/Application/Controller/Administration/ParametersController.php:94]: should not be true by default ?

Comment (adr-mo) [reply src/Centreon/Application/Controller/Administration/ParametersController.php]: you are totally right, this has been fixed

**Review by kduret — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Comment by adr-mo** [comment]
Force pushed the branch to rebase regarding develop

**Comment by adr-mo** [comment]
New commit handle so that ACK on Resources handles the force_active_checks param

**Pull request closed without merging**